src/client/kotlin/me/owdding/skyocean/features/inventory/buttons/InvButton.kt

```diff
@@ -0,0 +1,106 @@
+package me.owdding.skyocean.features.inventory.buttons
+
+import earth.terrarium.olympus.client.components.base.renderer.WidgetRenderer
+import earth.terrarium.olympus.client.components.base.renderer.WidgetRendererContext
+import earth.terrarium.olympus.client.components.buttons.Button
+import me.owdding.skyocean.SkyOcean.minecraft
+import me.owdding.skyocean.config.features.inventory.ButtonConfig
+import net.minecraft.client.gui.GuiGraphics
+import net.minecraft.client.gui.screens.Screen
+import net.minecraft.client.renderer.RenderType
+import net.minecraft.core.registries.BuiltInRegistries
+import net.minecraft.resources.ResourceLocation
+import net.minecraft.world.item.Items
+import tech.thatgravyboat.skyblockapi.api.remote.RepoItemsAPI
+import tech.thatgravyboat.skyblockapi.utils.extentions.pushPop
+import tech.thatgravyboat.skyblockapi.utils.text.TextProperties.stripped
+import kotlin.jvm.optionals.getOrNull
+
+class InvButton(
+    val button: ButtonConfig,
+    val rowIndex: Int,
+    val bottom: Boolean,
+    val screen: Screen,
+    val index: Int,
+    val baseX: Int,
+    val baseY: Int,
+    val baseWidth: Int,
+    val baseHeight: Int
+) : Button() {
+    var highlight = false
+    fun renderButtons(graphics: GuiGraphics, mouseX: Int, mouseY: Int, partialTicks: Float) {
+        graphics.pushPop {
+            val sprite = if (bottom) {
+                SELECTED_BOTTOM_TABS[rowIndex]
+            } else {
+                SELECTED_TOP_TABS[rowIndex]
+            }
+            renderPrevious(graphics, mouseX, mouseY, partialTicks, sprite)
+            val itemX = baseWidth / 2 - 8 + this@InvButton.x
+            val itemY = if (bottom) {
+                baseHeight  + this@InvButton.y - (baseWidth / 2) - 8
+            } else {
+                baseWidth / 2 - 8 + this@InvButton.y
+            }
+            val stack = if (button.item.contains(":")) {
+                BuiltInRegistries.ITEM.get(
+                    ResourceLocation.bySeparator(button.item, ':'))?.getOrNull()?.value()?.defaultInstance ?: Items.BARRIER.defaultInstance
+            } else {
+                RepoItemsAPI.getItem(button.item)
+            }
```

Comment:
or smth like that, dont think the distinction between skyblock and vanilla is necessary here

```suggestion
            val stack = BuiltInRegistries.ITEM.get(ResourceLocation.bySeparator(button.item, ':'))?.getOrNull()?.value()?.defaultInstance 
                ?: RepoItemsAPI.getItem(button.item)
                ?: Items.BARRIER.defaultInstance
```